akka-remote-tests/src/test/scala/akka/remote/artery/ArteryFailedToBindSpec.scala

```diff
@@ -17,7 +17,13 @@ class ArteryFailedToBindSpec extends AnyWordSpec with Matchers {
 
   "an ActorSystem" must {
     "not start if port is taken" in {
-      val port = SocketUtil.temporaryLocalPort(true)
+
+      // this test is tweaked in Jenkins CI by passing -Dakka.remote.artery.transport
+      // therefore we must decide whether to use UDP or not based on the runtime config
+      val arterySettings = ArterySettings(ConfigFactory.load().getConfig("akka.remote.artery"))
```

Comment:
better to get the settings from the actor system e.g. system.settings.config

Reply:
That would be too late. We need to know if we are configuring it for UDP or not before we start the Actor System. 

On line 25, I use this info to 'reserve' a port. That port is then added to the config that will be used by the Actor System.

Reply:
Btw, I could have used System.getProperties, but chose to load the whole config in order to compare using `arterySettings.Transport == ArterySettings.AeronUpd`. That gives us some protection against future chagnes (although I don't think this will change).

Reply:
Ah doh! Gotcha